ktor-utils/posix/src/io/ktor/util/CryptoNative.kt

```diff
@@ -4,12 +4,37 @@
 
 package io.ktor.util
 
+import kotlinx.cinterop.*
+import platform.Security.*
+
+private const val NONCE_SIZE_IN_BYTES = 16
 
 /**
- * Generates a nonce string. Could block if the system's entropy source is empty
+ * Generates a nonce string 16 characters long. Could block if the system's entropy source is empty
  */
 @InternalAPI
-actual fun generateNonce(): String = error("[generateNonce] is not supported on iOS")
+actual fun generateNonce(): String {
+    return hex(generateNonceByteArray())
+}
+
+private fun generateNonceByteArray(): ByteArray {
+    memScoped {
+        val array = allocArray<ByteVarOf<Byte>>(NONCE_SIZE_IN_BYTES)
+        val status = SecRandomCopyBytes(
+            kSecRandomDefault,
```

Comment:
What is it's threading model? Can we use it outside of the main event queue? If yes, can we do it concurrently?

Reply:
It can be called from outside the main event queue and also concurrently. Should I provide a sample for this?